python/graphstorm/model/node_glem.py

```diff
@@ -264,9 +264,11 @@ def toggle(self, part='lm', data=None):
             self.training_lm = True
             self.toggle_params('gnn', True)
             self.toggle_params('lm', False)
-            # when training lm, do not use the cached LM
-            self.lm.unfreeze_input_encoder()
+            # when training lm: gnn still uses the cached LM embeddings from previous iteration
+            # hence we don't unfreeze nor update the cache
         elif part == 'gnn':
+            # clear previous cache
+            self.lm.unfreeze_input_encoder()
```

Comment:
if we train GNN part, why do we unfreeze the input encoder?

Reply:
It is later frozen (line 276). This unfreeze call is to clear the outdated LM cache that was used for training LM from the previous iteration.